pkg/controller/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -721,33 +727,109 @@ func TestSyncMurStatusWithUserAccountStatuses(t *testing.T) {
-	s := apiScheme(t)
-	mur := murtest.NewMasterUserRecord(t, "john",
-		murtest.Finalizer("finalizer.toolchain.dev.openshift.com"),
-		murtest.ToBeDeleted())
-	userAcc := uatest.NewUserAccountFromMur(mur)
+	t.Run("success", func(t *testing.T) {
+		// given
+		defer counter.Reset()
+		InitializeCounter(t, 2, UserAccountsForCluster(test.MemberClusterName, 2))
+		logf.SetLogger(zap.New(zap.UseDevMode(true)))
+		s := apiScheme(t)
+		mur := murtest.NewMasterUserRecord(t, "john",
+			murtest.ToBeDeleted())
+		userAcc := uatest.NewUserAccountFromMur(mur)
 
-	memberClient := test.NewFakeClient(t, userAcc)
-	hostClient := test.NewFakeClient(t, mur)
+		memberClient := test.NewFakeClient(t, userAcc)
+		hostClient := test.NewFakeClient(t, mur)
 
-	cntrl := newController(t, hostClient, s, NewGetMemberCluster(true, v1.ConditionTrue),
-		ClusterClient(test.MemberClusterName, memberClient))
+		cntrl := newController(t, hostClient, s, NewGetMemberCluster(true, v1.ConditionTrue),
+			ClusterClient(test.MemberClusterName, memberClient))
 
-	// when
-	_, err := cntrl.Reconcile(newMurRequest(mur))
+		// when
+		result1, err1 := cntrl.Reconcile(newMurRequest(mur))
+		require.NoError(t, err1)
+		assert.True(t, result1.Requeue)
+		assert.Equal(t, int64(result1.RequeueAfter), int64(10*time.Second))
 
-	// then
-	require.NoError(t, err)
+		result2, err2 := cntrl.Reconcile(newMurRequest(mur))
 
-	uatest.AssertThatUserAccount(t, "john", memberClient).
-		DoesNotExist()
-	murtest.AssertThatMasterUserRecord(t, "john", hostClient).
-		DoesNotHaveFinalizer()
-	AssertThatCounterHas(t, 1, UserAccountsForCluster(test.MemberClusterName, 1))
+		// then
+		require.Empty(t, result2)
+		require.NoError(t, err2)
+
+		uatest.AssertThatUserAccount(t, "john", memberClient).
+			DoesNotExist()
+		murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+			DoesNotHaveFinalizer()
+		AssertThatCounterHas(t, 1, UserAccountsForCluster(test.MemberClusterName, 1))
+	})
+
+	t.Run("test wait for UserAccount deletion in progress before removing MUR finalizer", func(t *testing.T) {
+		// given
+		defer counter.Reset()
+		InitializeCounter(t, 2, UserAccountsForCluster(test.MemberClusterName, 2))
+		logf.SetLogger(zap.New(zap.UseDevMode(true)))
+		s := apiScheme(t)
+		mur := murtest.NewMasterUserRecord(t, "john-wait-for-ua",
+			murtest.ToBeDeleted())
+		userAcc := uatest.NewUserAccountFromMur(mur)
+		// set deletion timestamp to indicate UserAccount deletion is in progress
+		userAcc.DeletionTimestamp = &metav1.Time{Time: time.Now().Add(-5 * time.Second)}
+
+		hostClient := test.NewFakeClient(t, mur)
+		memberClient := test.NewFakeClient(t, userAcc)
+
+		cntrl := newController(t, hostClient, s, NewGetMemberCluster(true, v1.ConditionTrue),
+			ClusterClient(test.MemberClusterName, memberClient))
+
+		// when
+		result1, err := cntrl.Reconcile(newMurRequest(mur))
+		require.NoError(t, err)
+		assert.True(t, result1.Requeue)
+		assert.Equal(t, int64(result1.RequeueAfter), int64(10*time.Second))
```

Comment:
the same also here

Reply:
Done in d203faf29d69fbe2b0a700a162d64ac3ba223dea